### PR TITLE
Close login ports and send failure html

### PR DESCRIPTION
**Pull request by Sheikah45**
Ensures that the websockets are closed if the user hasn't logged in within 5 minutes of requesting the login, making the ports available for reuse and are not left open indefinitely.

Also removes the statefulness from OAuthValuesReceiver since it was only being used for testing

### CODE REVIEW

File: src/main/java/com/faforever/client/login/OAuthValuesReceiver.java [hunk 1]
```diff
@@ -38,17 +35,14 @@
...
 @Slf4j
 public class OAuthValuesReceiver {
 
   private static final Pattern CODE_PATTERN = Pattern.compile("code=([^ &]+)");
   private static final Pattern STATE_PATTERN = Pattern.compile("state=([^ &]+)");
   private static final List<String> ALLOWED_HOSTS = List.of("localhost");
-
-  @VisibleForTesting
-  CompletableFuture<Integer> listenPort = new CompletableFuture<>();
+  private static final int MILLISECONDS_TO_WAIT = 300000;
```
Comment (Brutus5000) [inline src/main/java/com/faforever/client/login/OAuthValuesReceiver.java]: Is this still used?

**Review by Brutus5000 — COMMENTED** [review]
5 minutes seems **very** short. I'd go for 15.

File: src/main/java/com/faforever/client/config/ClientProperties.java [hunk 2]
```diff
@@ -224,5 +224,6 @@ public void updateFromEndpoint(ServerEndpoints serverEndpoints) {
     private URI redirectUri = URI.create("");
     private String clientId;
     private String scopes;
+    private int timeout;
```
Comment (Brutus5000) [inline src/main/java/com/faforever/client/config/ClientProperties.java]: nobody knows this is supposed to be milliseconds

Comment (Sheikah45) [reply src/main/java/com/faforever/client/config/ClientProperties.java]: Good point

File: src/main/java/com/faforever/client/login/OAuthValuesReceiver.java (hunk 1 above)
Comment (Sheikah45) [reply src/main/java/com/faforever/client/login/OAuthValuesReceiver.java]: Nope can be removed

**Comment by Sheikah45** [comment]
> 5 minutes seems **very** short. I'd go for 15.

Must be a slow clicker XD

**Pull request merged**